packages/common/src/search/utils.ts

```diff
@@ -413,3 +381,32 @@ export function serializeStringOrArray(
   }
   return q;
 }
+
+/**
+ * Create a `.next()` function for a type
+ * @param request
+ * @param nextStart
+ * @param total
+ * @param fn
+ * @returns
+ */
+export function getNextFunction<T>(
+  request: ISearchOptions,
+  nextStart: number,
+  total: number,
+  fn: (r: ISearchOptions) => Promise<ISearchResponse<T>>
+): () => Promise<ISearchResponse<T>> {
+  const clonedRequest = cloneObject(request);
+
+  // Can't clone a UserSession
+  clonedRequest.authentication = request.authentication;
```

Comment:
Maybe we can do something like:

```suggestion
  clonedRequest.authentication = UserSession.desrialize(request.authentication.serialize());
```